### PR TITLE
update for air purification unit

### PR DESCRIPTION
**Pull request by Karthi8883**
updated telemetry, fan and sub fields

### CODE REVIEW

**Comment by tasodorff** [comment]
validator errors

File: ontology/yaml/resources/subfields/subfields.yaml [hunk 1]
```diff
@@ -37,6 +37,7 @@ component:
   dial: "Adjustment device (e.g. setpoint dial)."
   dimmer: "A device which can reduce lighting output of a lamp (thus 'dimming' the light)."
   dishwasher: "Device that washes dishes."
+  door: "Door is a barrier (hinged or otherwise) which allows entry or exit from a space or enclosure"
```
Comment (tasodorff) [inline ontology/yaml/resources/subfields/subfields.yaml:40]: Remove "Door is"

**Review by tasodorff — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 2]
```diff
@@ -301,6 +301,22 @@ FAN_SS_VSC_EFC_FDPM:
   - EFC
   - FDPM
 
+FAN_SS_VSC_OA:
```
[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:304]: This is a HVAC Fan

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 3]
```diff
@@ -301,6 +301,22 @@ FAN_SS_VSC_EFC_FDPM:
...
+  implements:
+  - FAN
+  - SS
+  - VSC
+  - OA
+  opt_uses:
+  - filter_alarm
+  - differential_pressure_sensor
+  - differential_pressure_setpoint
+  - run_mode
```
[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:316]: Updated

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 4]
```diff
@@ -301,6 +301,22 @@ FAN_SS_VSC_EFC_FDPM:
...
+  - door_open_alarm
```
[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:317]: Door of the Fan enclosure.

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 5]
```diff
@@ -301,6 +301,22 @@ FAN_SS_VSC_EFC_FDPM:
...
   - FDPM
 
+FAN_SS_VSC_OA:
+  description: "Fan with start/stop status, speed control with fire, EC fan and filter monitoring for purification"
+  is_canonical: true
+  implements:
+  - FAN
+  - SS
+  - VSC
+  - OA
```
[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:318]: Updated

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (tasodorff) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:324]: why is it relevant to the fan? should this instead be modeled as a door? we have a door concept in the facilities namespace. i dont see the connection to the fan except in that they are located in the same general space.

[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml]: Noted, will not model door here.

File: ontology/yaml/resources/fields/telemetry_fields.yaml [hunk 6]
```diff
@@ -1487,6 +1487,10 @@ literals:
   - ON
   - OFF
 
+# Addition of telemetry for door alarm
+- door_open_alarm:
```
Comment (tasodorff) [inline ontology/yaml/resources/fields/telemetry_fields.yaml:1491]: is this still needed?

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 7]
```diff
@@ -301,6 +301,22 @@ FAN_SS_VSC_EFC_FDPM:
...
+  opt_uses:
+  - filter_alarm
+  - differential_pressure_sensor
+  - differential_pressure_setpoint
+  - run_mode
+  - door_open_alarm
+  - fire_alarm
```
[the remark this answers is not in the document]
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml:325]: removed fire alarm and abstract added for FDPSM4X for filter pressure monitoring

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml [hunk 8]
```diff
@@ -308,6 +308,17 @@ FAN_SS_VSC_EFC_FDPM:
   - EFC
   - FDPM
 
+FAN_SS_VSC_WDPC_STM_FDPSM4X:
+  description: "Fan with start/stop status, speed control with fire, EC fan and filter status monitoring for purification"
```
Comment (tasodorff) [inline ontology/yaml/resources/HVAC/entity_types/FAN.yaml:312]: where is the fire data? either modify the point list or modify the description

**Review by tasodorff — CHANGES_REQUESTED** [review]

File: ontology/yaml/resources/HVAC/entity_types/FAN.yaml (hunk 8 above)
Comment (Karthi8883) [reply ontology/yaml/resources/HVAC/entity_types/FAN.yaml]: Modified the description

**Review by tasodorff — APPROVED** [review]

**Comment by mschulze17** [comment]
@cstirdivant yours to merge

**Pull request merged**